### PR TITLE
[DOCS] Adds new dynamic machine learning settings

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/33961 and https://github.com/elastic/elasticsearch/issues/33565

This PR adds documentation for the new dynamic settings.

### CODE REVIEW

**Comment by lcawl** [comment]
@droberts195 and @benwtrent: One thing that wasn't clear to me was whether the percentage of date/order errors is relative to a subset of the data (e.g. per bucket). If we can clarify that in the description, that'd be great.

File: docs/reference/settings/ml-settings.asciidoc [hunk 1]
```diff
@@ -57,8 +58,33 @@ that is greater than this setting value, an error occurs. Existing jobs are not
...
+==== Advanced machine learning settings
+
+These settings are for advanced use cases; the default values are generally 
+sufficient:
+
+`xpack.ml.max_anomaly_records`:: (<<cluster-update-settings,Dynamic>>) 
+The maximum number of records that are output per bucket. The default value is 
+`500`.
+
+`xpack.ml.max_percent_date_errors`:: (<<cluster-update-settings,Dynamic>>)
```
Comment (droberts195) [inline docs/reference/settings/ml-settings.asciidoc]: It turns out that this setting and `xpack.ml.max_percent_out_of_order_errors` are never used anywhere.  (They used to be used in the old Prelert product.)

So they should be removed completely from the docs and code.

**Comment by lcawl** [comment]
Thanks @droberts195, I've removed those two settings.

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**